CRITICAL - 2023-10-30 01:12:20 --> Undefined variable $datos
in APPPATH\Views\dashboard_bibliotecario.php on line 55.
 1 APPPATH\Views\dashboard_bibliotecario.php(55): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Undefined variable $datos', 'C:\\xampp\\htdocs\\Proyectofinal\\pp2023_Biblioteca_virtual\\app\\Views\\dashboard_bibliotecario.php', 55)
 2 SYSTEMPATH\View\View.php(224): include('C:\\xampp\\htdocs\\Proyectofinal\\pp2023_Biblioteca_virtual\\app\\Views\\dashboard_bibliotecario.php')
 3 SYSTEMPATH\View\View.php(227): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 4 SYSTEMPATH\Common.php(1180): CodeIgniter\View\View->render('dashboard_bibliotecario', [], true)
 5 APPPATH\Controllers\Dashboard.php(20): view('dashboard_bibliotecario')
 6 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\Dashboard->dashbiblio()
 7 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Dashboard))
 8 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
CRITICAL - 2023-10-30 01:13:22 --> Undefined variable $datos
in APPPATH\Views\dashboard_bibliotecario.php on line 55.
 1 APPPATH\Views\dashboard_bibliotecario.php(55): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Undefined variable $datos', 'C:\\xampp\\htdocs\\Proyectofinal\\pp2023_Biblioteca_virtual\\app\\Views\\dashboard_bibliotecario.php', 55)
 2 SYSTEMPATH\View\View.php(224): include('C:\\xampp\\htdocs\\Proyectofinal\\pp2023_Biblioteca_virtual\\app\\Views\\dashboard_bibliotecario.php')
 3 SYSTEMPATH\View\View.php(227): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 4 SYSTEMPATH\Common.php(1180): CodeIgniter\View\View->render('dashboard_bibliotecario', [], true)
 5 APPPATH\Controllers\Dashboard.php(20): view('dashboard_bibliotecario')
 6 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\Dashboard->dashbiblio()
 7 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Dashboard))
 8 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:17:14 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:17:14 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'2\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'2\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'2\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:17:40 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:17:40 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:18:08 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:18:08 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:18:13 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:18:13 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'4\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'4\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'4\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:18:16 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:18:16 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'10\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'10\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'10\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:18:19 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:18:19 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'12\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'12\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'12\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:18:26 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:18:26 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'12\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'12\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'12\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:24:13 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:24:13 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-06\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:24:24 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('4', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:24:24 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'Mario\', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-11\', `fecha_devolucion` = \'2023-10-10\'
WHERE `prestamos`.`prestamo_id` IN (\'4\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'Mario\', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-11\', `fecha_devolucion` = \'2023-10-10\'
WHERE `prestamos`.`prestamo_id` IN (\'4\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'Mario\', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-11\', `fecha_devolucion` = \'2023-10-10\'
WHERE `prestamos`.`prestamo_id` IN (\'4\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('4', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:24:30 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:24:30 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'2\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'2\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'2\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:24:40 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:24:40 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'2\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'2\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'2\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:25:16 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`libros`, CONSTRAINT `libros_ibfk_1` FOREIGN KEY (`autor_id`) REFERENCES `autores` (`autor_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `libros`...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `libros`...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `libros`...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `libros`...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerLibros.php(99): CodeIgniter\Model->update('12', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerLibros->actualizarLibro()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerLibros))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:25:16 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`libros`, CONSTRAINT `libros_ibfk_1` FOREIGN KEY (`autor_id`) REFERENCES `autores` (`autor_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `libros` SET `titulo` = \'prueba2\', `autor_id` = \'\', `categoria_id` = \'2\', `descripcion` = \'prueba2\', `anio_publicacion` = \'2023\', `cantidad_disponible` = \'45\'
WHERE `libros`.`libro_id` IN (\'12\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `libros` SET `titulo` = \'prueba2\', `autor_id` = \'\', `categoria_id` = \'2\', `descripcion` = \'prueba2\', `anio_publicacion` = \'2023\', `cantidad_disponible` = \'45\'
WHERE `libros`.`libro_id` IN (\'12\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `libros` SET `titulo` = \'prueba2\', `autor_id` = \'\', `categoria_id` = \'2\', `descripcion` = \'prueba2\', `anio_publicacion` = \'2023\', `cantidad_disponible` = \'45\'
WHERE `libros`.`libro_id` IN (\'12\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `libros` SET `titulo` = :titulo:, `autor_id` = :autor_id:, `categoria_id` = :categoria_id:, `descripcion` = :descripcion:, `anio_publicacion` = :anio_publicacion:, `cantidad_disponible` = :cantidad_disponible:
WHERE `libros`.`libro_id` IN :libros.libro_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerLibros.php(99): CodeIgniter\Model->update('12', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerLibros->actualizarLibro()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerLibros))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:25:42 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:25:42 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-16\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-16\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-16\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:26:09 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:26:09 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-16\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-16\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'José \', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-29\', `fecha_devolucion` = \'2023-11-16\'
WHERE `prestamos`.`prestamo_id` IN (\'7\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('7', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:29:20 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`libros`, CONSTRAINT `libros_ibfk_2` FOREIGN KEY (`categoria_id`) REFERENCES `categorias` (`categoria_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `libros`...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `libros`...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `libros`...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `libros`...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerLibros.php(99): CodeIgniter\Model->update('1', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerLibros->actualizarLibro()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerLibros))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:29:20 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`libros`, CONSTRAINT `libros_ibfk_2` FOREIGN KEY (`categoria_id`) REFERENCES `categorias` (`categoria_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `libros` SET `titulo` = \'Cien años de soledad\', `autor_id` = \'3\', `categoria_id` = \'\', `descripcion` = \'Una obra maestra de la literatura latinoamericana.\', `anio_publicacion` = \'1967\', `cantidad_disponible` = \'10\'
WHERE `libros`.`libro_id` IN (\'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `libros` SET `titulo` = \'Cien años de soledad\', `autor_id` = \'3\', `categoria_id` = \'\', `descripcion` = \'Una obra maestra de la literatura latinoamericana.\', `anio_publicacion` = \'1967\', `cantidad_disponible` = \'10\'
WHERE `libros`.`libro_id` IN (\'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `libros` SET `titulo` = \'Cien años de soledad\', `autor_id` = \'3\', `categoria_id` = \'\', `descripcion` = \'Una obra maestra de la literatura latinoamericana.\', `anio_publicacion` = \'1967\', `cantidad_disponible` = \'10\'
WHERE `libros`.`libro_id` IN (\'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `libros` SET `titulo` = :titulo:, `autor_id` = :autor_id:, `categoria_id` = :categoria_id:, `descripcion` = :descripcion:, `anio_publicacion` = :anio_publicacion:, `cantidad_disponible` = :cantidad_disponible:
WHERE `libros`.`libro_id` IN :libros.libro_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerLibros.php(99): CodeIgniter\Model->update('1', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerLibros->actualizarLibro()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerLibros))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:30:11 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`libros`, CONSTRAINT `libros_ibfk_2` FOREIGN KEY (`categoria_id`) REFERENCES `categorias` (`categoria_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `libros`...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `libros`...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `libros`...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `libros`...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerLibros.php(99): CodeIgniter\Model->update('1', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerLibros->actualizarLibro()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerLibros))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:30:11 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`libros`, CONSTRAINT `libros_ibfk_2` FOREIGN KEY (`categoria_id`) REFERENCES `categorias` (`categoria_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `libros` SET `titulo` = \'Cien años de soledad\', `autor_id` = \'1\', `categoria_id` = \'\', `descripcion` = \'Una obra maestra de la literatura latinoamericana.\', `anio_publicacion` = \'1967\', `cantidad_disponible` = \'10\'
WHERE `libros`.`libro_id` IN (\'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `libros` SET `titulo` = \'Cien años de soledad\', `autor_id` = \'1\', `categoria_id` = \'\', `descripcion` = \'Una obra maestra de la literatura latinoamericana.\', `anio_publicacion` = \'1967\', `cantidad_disponible` = \'10\'
WHERE `libros`.`libro_id` IN (\'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `libros` SET `titulo` = \'Cien años de soledad\', `autor_id` = \'1\', `categoria_id` = \'\', `descripcion` = \'Una obra maestra de la literatura latinoamericana.\', `anio_publicacion` = \'1967\', `cantidad_disponible` = \'10\'
WHERE `libros`.`libro_id` IN (\'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `libros` SET `titulo` = :titulo:, `autor_id` = :autor_id:, `categoria_id` = :categoria_id:, `descripcion` = :descripcion:, `anio_publicacion` = :anio_publicacion:, `cantidad_disponible` = :cantidad_disponible:
WHERE `libros`.`libro_id` IN :libros.libro_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerLibros.php(99): CodeIgniter\Model->update('1', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerLibros->actualizarLibro()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerLibros))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:32:56 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`libros`, CONSTRAINT `libros_ibfk_1` FOREIGN KEY (`autor_id`) REFERENCES `autores` (`autor_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `libros`...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `libros`...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `libros`...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `libros`...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerLibros.php(99): CodeIgniter\Model->update('1', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerLibros->actualizarLibro()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerLibros))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:32:56 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`libros`, CONSTRAINT `libros_ibfk_1` FOREIGN KEY (`autor_id`) REFERENCES `autores` (`autor_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `libros` SET `titulo` = \'Cien años de soledad XX\', `autor_id` = \'\', `categoria_id` = \'\', `descripcion` = \'Una obra maestra de la literatura latinoamericana.\', `anio_publicacion` = \'1967\', `cantidad_disponible` = \'10\'
WHERE `libros`.`libro_id` IN (\'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `libros` SET `titulo` = \'Cien años de soledad XX\', `autor_id` = \'\', `categoria_id` = \'\', `descripcion` = \'Una obra maestra de la literatura latinoamericana.\', `anio_publicacion` = \'1967\', `cantidad_disponible` = \'10\'
WHERE `libros`.`libro_id` IN (\'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `libros` SET `titulo` = \'Cien años de soledad XX\', `autor_id` = \'\', `categoria_id` = \'\', `descripcion` = \'Una obra maestra de la literatura latinoamericana.\', `anio_publicacion` = \'1967\', `cantidad_disponible` = \'10\'
WHERE `libros`.`libro_id` IN (\'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `libros` SET `titulo` = :titulo:, `autor_id` = :autor_id:, `categoria_id` = :categoria_id:, `descripcion` = :descripcion:, `anio_publicacion` = :anio_publicacion:, `cantidad_disponible` = :cantidad_disponible:
WHERE `libros`.`libro_id` IN :libros.libro_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerLibros.php(99): CodeIgniter\Model->update('1', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerLibros->actualizarLibro()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerLibros))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:33:39 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:33:39 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:33:43 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:33:43 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:33:46 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:33:46 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'4\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'4\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'4\', `fecha_prestamo` = \'2023-10-02\', `fecha_devolucion` = \'2023-10-20\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2023-10-30 03:34:03 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`)) in C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php:295
Stack trace:
#0 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestam...', 0)
#1 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestam...')
#2 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestam...')
#3 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestam...', Array, false)
#4 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
#5 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\BaseModel.php(943): CodeIgniter\Model->doUpdate(Array, Array)
#6 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\Model.php(756): CodeIgniter\BaseModel->update(Array, Array)
#7 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\app\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', Array)
#8 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
#9 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
#10 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\system\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Proyectofinal\pp2023_Biblioteca_virtual\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 {main}
CRITICAL - 2023-10-30 03:34:03 --> Cannot add or update a child row: a foreign key constraint fails (`pp2023_biblioteca_virtual`.`prestamos`, CONSTRAINT `prestamos_ibfk_1` FOREIGN KEY (`usuario_id`) REFERENCES `usuarios` (`usuario_id`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 295.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(295): mysqli->query('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-19\', `fecha_devolucion` = \'2023-10-26\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(692): CodeIgniter\Database\MySQLi\Connection->execute('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-19\', `fecha_devolucion` = \'2023-10-26\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(606): CodeIgniter\Database\BaseConnection->simpleQuery('UPDATE `prestamos` SET `usuario_id` = \'Juan\', `libro_id` = \'1\', `fecha_prestamo` = \'2023-10-19\', `fecha_devolucion` = \'2023-10-26\'
WHERE `prestamos`.`prestamo_id` IN (\'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2476): CodeIgniter\Database\BaseConnection->query('UPDATE `prestamos` SET `usuario_id` = :usuario_id:, `libro_id` = :libro_id:, `fecha_prestamo` = :fecha_prestamo:, `fecha_devolucion` = :fecha_devolucion:
WHERE `prestamos`.`prestamo_id` IN :prestamos.prestamo_id:', [...], false)
 5 SYSTEMPATH\Model.php(396): CodeIgniter\Database\BaseBuilder->update()
 6 SYSTEMPATH\BaseModel.php(943): CodeIgniter\Model->doUpdate([...], [...])
 7 SYSTEMPATH\Model.php(756): CodeIgniter\BaseModel->update([...], [...])
 8 APPPATH\Controllers\ControllerPrestamos.php(90): CodeIgniter\Model->update('1', [...])
 9 SYSTEMPATH\CodeIgniter.php(932): App\Controllers\ControllerPrestamos->actualizarPrestamo()
10 SYSTEMPATH\CodeIgniter.php(497): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\ControllerPrestamos))
11 SYSTEMPATH\CodeIgniter.php(366): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
